src/Storage/Record.php

```diff
@@ -0,0 +1,174 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\entity_hierarchy\Storage;
+
+use Drupal\Core\Entity\ContentEntityInterface;
+
+/**
+ * Defines a database record for a hierarchical query result.
+ */
+class Record {
+
+  protected int $id;
+
+  protected ?int $revisionId = NULL;
+
+  protected ?int $targetId;
+
+  protected int $weight;
+
+  protected int $depth;
```

Comment:
should we initialize these to a value? both the methods return `:int` not `:?int` so do we need a default value?